.github/workflows/bench-pr-comment.yml

```diff
@@ -1,71 +1,118 @@
+      default-benches:
         type: string
-      bench:
         required: true
+      # Whitespace-separated list of default env vars, set regardless of comment body
+      default-env:
         type: string
+        required: false
 
-# TODO: Configure env vars via the `!benchmark` comment body instead of `bench.env`
 jobs:
-  cpu-benchmark:
-    name: Run CPU benchmark
-    runs-on: ${{ inputs.runner }}
+  setup:
+    name: Set up benchmark parameters
+    runs-on: ubuntu-latest
+    if:
+      github.event.issue.pull_request
+      && github.event.issue.state == 'open'
+      && contains(github.event.comment.body, '!benchmark')
+      && (github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'OWNER')
+    outputs:
+      # Benches specified by `--bench <BENCH>` repeated for each bench
+      benches: ${{ steps.bench-params.outputs.benches }}
+      # Features specified by `--features <FEATURES>`
+      features: ${{ steps.bench-params.outputs.features }}
+      # Env vars specified by `ENV_VAR=<VAR>`, starting on the second line of the `issue_comment` input
+      # Separated by whitespace but ideally newlines for readability
+      env-vars: ${{ steps.bench-params.outputs.env-vars }}
+      # Flag to denote the `cuda` feature is active, which means we need a self-hosted GPU runner
+      cuda: ${{ steps.bench-params.outputs.cuda }}
+
     steps:
-      - uses: actions/checkout@v4
-        with:
-          repository: lurk-lab/ci-workflows
-      - uses: ./.github/actions/gpu-setup
-        with:
-          gpu-framework: 'cuda'
-      - uses: ./.github/actions/ci-env
-      - uses: actions/checkout@v4
-      - name: Checkout PR branch
-        run: gh pr checkout $PR_NUMBER
-        env:
-          GH_TOKEN: ${{ github.token }}
-          PR_NUMBER: ${{ github.event.issue.number }}
-      # Install dependencies
-      - name: Install dependencies
-        run: sudo apt-get install -y pkg-config libssl-dev
-      - uses: dtolnay/rust-toolchain@stable
-      - uses: Swatinem/rust-cache@v2
-      - name: Set output type
+      - name: Parse PR comment body
+        id: bench-params
         run: |
-          echo "LURK_BENCH_OUTPUT=pr-comment" >> $GITHUB_ENV
-      - uses: cardinalby/export-env-action@v2
-        with:
-          envFile:
-            'benches/bench.env'
-      # Run the comparative benchmark and comment output on the PR
-      - uses: boa-dev/criterion-compare-action@v3
-        with:
-          # Note: Removing `benchName` causes `criterion` errors: https://github.com/boa-dev/criterion-compare-action#troubleshooting
-          # Optional. Compare only this benchmark target
-          benchName: "${{ inputs.bench }}"
-          # Needed. The name of the branch to compare with
-          branchName: ${{ github.ref_name }}
+          printf '${{ github.event.comment.body }}' > comment.txt
+          BENCH_COMMAND=$(head -n 1 comment.txt)
+          echo $BENCH_COMMAND
```

Comment:
Consider quoting `echo "$BENCH_COMMAND"`, here and elsewhere.

Reply:
Could you explain/point me to some resources on why we always want to double-quote env vars with `echo`? Does this extend to all usage of env vars and/or GitHub variables?

Single vs. double quotes and parsing env vars into properly-quoted JSON arrays for the `fromJSON` function was by far the hardest part of this PR, so I'd like to understand it better :pray:

Reply:
The ultimate reference for me is "Learning the Bash Shell", chapter 8, section command-line processing, and its figure.